### PR TITLE
Fix GDB

### PR DESCRIPTION
**Pull request by saagarjha**
Newer versions of GDB read from /proc/pid/task.

### CODE REVIEW

File: kernel/ptrace.c [hunk 1]
```diff
@@ -119,12 +119,7 @@ dword_t sys_ptrace(dword_t request, dword_t pid, addr_t addr, dword_t data) {
             struct task *child = find_child(pid);
             if (!child) return _EPERM;
 
-            read_wrlock(&child->mem->lock);
```
Comment (tbodt) [inline kernel/ptrace.c:122]: why change this? was it broken?

Comment (tbodt) [reply kernel/ptrace.c:122]: oh i forgot about the /proc/pid/mem, never mind

File: kernel/user.c [hunk 2]
```diff
@@ -61,7 +68,7 @@ int user_read_string(addr_t addr, char *buf, size_t max) {
     read_wrlock(&current->mem->lock);
     size_t i = 0;
     while (i < max) {
-        if (__user_read_task(current, addr + i, &buf[i], sizeof(buf[i]))) {
+        if (__user_read_task(current, addr + i, &buf[i], sizeof(buf[i])), false) {
```
Comment (tbodt) [inline kernel/user.c:71]: uh does this compile? no parameter was added to __user_read_task

Comment (saagarjha) [reply kernel/user.c:71]: You're reading it wrong, this is using the comma operator

Comment (tbodt) [reply kernel/user.c]: this seems like a bad idea, if you're going to ignore the return value, why put the call in the if at all? very easy to read wrong as you've seen

Comment (saagarjha) [reply kernel/user.c]: I think you're still reading it wrong

**Pull request merged**